datasketches/src/frequencies/sketch.rs

```diff
@@ -0,0 +1,408 @@
+    }
+
+    /// Returns frequent items using a custom threshold.
+    pub fn get_frequent_items_with_threshold(
+        &self,
+        error_type: ErrorType,
+        threshold: i64,
+    ) -> Vec<Row<T>>
+    where
+        T: Clone,
+    {
+        let threshold = threshold.max(self.offset);
+        let mut rows = Vec::new();
+        for (item, count) in self.hash_map.iter() {
+            let lower = count;
+            let upper = count + self.offset;
+            let include = match error_type {
+                ErrorType::NoFalseNegatives => upper > threshold,
+                ErrorType::NoFalsePositives => lower > threshold,
+            };
+            if include {
+                rows.push(Row {
+                    item: item.clone(),
+                    estimate: upper,
+                    upper_bound: upper,
+                    lower_bound: lower,
+                });
+            }
+        }
+        rows.sort_by(|a, b| b.estimate.cmp(&a.estimate));
+        rows
+    }
+
+    /// Serializes this sketch into a byte vector using the provided serializer.
+    pub fn serialize_with<S: ItemsSerde<T>>(&self, serde: &S) -> Vec<u8>
+    where
+        T: Clone,
+    {
+        if self.is_empty() {
+            let mut out = vec![0u8; 8];
+            out[PREAMBLE_LONGS_BYTE] = PREAMBLE_LONGS_EMPTY;
+            out[SER_VER_BYTE] = SER_VER;
+            out[FAMILY_BYTE] = FAMILY_ID;
+            out[LG_MAX_MAP_SIZE_BYTE] = self.lg_max_map_size;
+            out[LG_CUR_MAP_SIZE_BYTE] = self.hash_map.get_lg_length();
+            out[FLAGS_BYTE] = EMPTY_FLAG_MASK;
+            return out;
+        }
+        let active_items = self.get_num_active_items();
+        let values = self.hash_map.get_active_values();
+        let keys = self.hash_map.get_active_keys();
+        let items_bytes = serde.serialize_items(&keys);
+        let total_bytes =
+            PREAMBLE_LONGS_NONEMPTY as usize * 8 + (active_items * 8) + items_bytes.len();
+        let mut out = vec![0u8; total_bytes];
+        out[PREAMBLE_LONGS_BYTE] = PREAMBLE_LONGS_NONEMPTY;
+        out[SER_VER_BYTE] = SER_VER;
+        out[FAMILY_BYTE] = FAMILY_ID;
+        out[LG_MAX_MAP_SIZE_BYTE] = self.lg_max_map_size;
+        out[LG_CUR_MAP_SIZE_BYTE] = self.hash_map.get_lg_length();
+        out[FLAGS_BYTE] = 0;
+        write_u32_le(&mut out, ACTIVE_ITEMS_INT, active_items as u32);
+        write_i64_le(&mut out, STREAM_WEIGHT_LONG, self.stream_weight);
+        write_i64_le(&mut out, OFFSET_LONG, self.offset);
+
+        let mut offset = PREAMBLE_LONGS_NONEMPTY as usize * 8;
+        for value in values {
+            write_i64_le(&mut out, offset, value);
+            offset += 8;
+        }
+        out[offset..offset + items_bytes.len()].copy_from_slice(&items_bytes);
+        out
+    }
+
+    /// Deserializes a sketch from bytes using the provided serializer.
+    pub fn deserialize_with<S: ItemsSerde<T>>(bytes: &[u8], serde: &S) -> Result<Self, SerdeError>
```

Comment:
```suggestion
    pub fn deserialize<S: ItemsSerde<T>>(bytes: &[u8], serde: &S) -> Result<Self, SerdeError>
```

Ditto. The default deserialize requires a policy param. So we don't need `_with` anyway.